frontcamp-patterns/components/newsapi/newsapi.decorator.js

```diff
@@ -0,0 +1,13 @@
+"use strict";
+
+import NewsApi from "./newsapi.module";
+
+export default class NewsApiDecorator extends NewsApi {
+    constructor(newsApiObject) {
+        super(newsApiObject.sourceApi, newsApiObject.keyApi, newsApiObject.containerApi, newsApiObject.documentObj);
```

Comment:
I still don't like this realization. It's not decorator  since you create new object. The idea of decorator that you should use method of original object but add some extra behavior for it.